### ISSUE TITLE
SymbolicRegression example results in overly complex expression

### ISSUE THREAD

**Issue by claforte** [wildart/Evolutionary.jl#78]
First, thanks for building and sharing such an exciting and promising package!

I tried your SymbolicRegression example: https://github.com/wildart/Evolutionary.jl/blob/master/examples/SymbolicRegression.ipynb

In section `1.3.1 Linear`, the solution found is much more complex than your example. It goes on for pages. ;-) (screenshot below)

Maybe `maxdepth` is no longer respected?

Thank you!

Christian

![image](https://user-images.githubusercontent.com/5090329/104837617-b96fbd00-5883-11eb-840a-da78756f4721.png)

**Comment by wildart** [comment]
> Maybe maxdepth is no longer respected?

Indeed. I wrote very basic tree mutation & crossover implementations without any constraints. More GP operations are needed.
